externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/RowSet.scala

```diff
@@ -24,15 +24,24 @@ import scala.collection.JavaConverters._
 import org.apache.hive.service.rpc.thrift._
 import org.apache.spark.sql.Row
 import org.apache.spark.sql.execution.HiveResult
+import org.apache.spark.sql.execution.HiveResult.TimeFormatters
 import org.apache.spark.sql.types._
 
 import org.apache.kyuubi.util.RowSetUtils._
 
 object RowSet {
 
+  private val timeUnrelatedDataTypes: Set[DataType] =
+    Set(BooleanType, FloatType, BinaryType, StringType)
+
   def toHiveString(valueAndType: (Any, DataType), nested: Boolean = false): String = {
     // compatible w/ Spark 3.1 and above
-    val timeFormatters = HiveResult.getTimeFormatters
+    val timeFormatters: TimeFormatters = valueAndType match {
```

Comment:
We may need to create a timeFormatters for a dataset instead of each cell.

Reply:
Refer to the implementation in spark: https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/RowSetUtils.scala#L37

Reply:
I think we can reuse it in the `toTRowSet` method

Reply:
Sure. Adopted.